### PR TITLE
[CIS-830] Image gallery audit

### PR DESCRIPTION
**Pull request by fortmarek**
## Description of the pull request

- Audit `ChatMessageImageGallery`
- Use it in attachment injector

After merging this, I can progress further on [this](https://github.com/GetStream/stream-chat-swift/pull/1003) PR.

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/Attachments/GalleryAttachmentViewInjector.swift [hunk 1]
```diff
@@ -14,19 +14,8 @@ public protocol GalleryContentViewDelegate: ChatMessageContentViewDelegate {
...
-    open lazy var galleryView: UIImageView = {
-        let imageView = EagerImageView()
-        imageView.contentMode = .scaleAspectFill
-        imageView.isUserInteractionEnabled = true
-        imageView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(handleTapOnImage)))
-        imageView.clipsToBounds = true
-        return imageView.withoutAutoresizingMaskConstraints
-    }()
+    public private(set) lazy var galleryView = _ChatMessageImageGallery<ExtraData>()
+        .withoutAutoresizingMaskConstraints
```
[the remark this answers is not in the document]
Comment (VojtaStavik) [reply Sources/StreamChatUI/ChatMessageList/Attachments/GalleryAttachmentViewInjector.swift:18]: I see ... all good then 👍

File: Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery.swift [hunk 2]
```diff
@@ -123,24 +89,17 @@ open class _ChatMessageImageGallery<ExtraData: ExtraDataTypes>: _View, ThemeProv
             itemPreview.isHidden = itemPreview.content == nil
         }
 
-        let visiblePreviewsCount = previews.filter { !$0.isHidden }.count
-        layouts.flatMap { $0 }.forEach { $0.isActive = false }
-        layouts[max(visiblePreviewsCount - 1, 0)].forEach { $0.isActive = true }
+        rightPreviewsContainerView.isHidden = rightPreviewsContainerView.subviews
+            .allSatisfy(\.isHidden)
```
[the remark this answers is not in the document]
Comment (fortmarek) [reply Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery.swift:95]: Yeah, I am not sure how `UIStackView` behaves in this scenario. But added the comment, anyways.

File: Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery+ImagePreview.swift [hunk 3]
```diff
@@ -69,17 +69,12 @@ extension _ChatMessageImageGallery {
         override open func updateContent() {
             let attachment = content
 
-            if let url = attachment?.payload?.imagePreviewURL {
-                loadingIndicator.isVisible = true
-                imageTask = loadImage(with: url, options: .shared, into: imageView, completion: { [weak self] _ in
+            loadingIndicator.isVisible = true
+            imageTask = imageView
+                .loadImage(from: attachment?.payload?.imagePreviewURL, resizeAutomatically: false) { [weak self] _ in
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery+ImagePreview.swift:74]: Why `resizeAutomatically: false` here? This is just the preview, it should be resized imo.

File: Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery.swift [hunk 4]
```diff
@@ -5,116 +5,81 @@
...
+        leftPreviewsContainerView.addArrangedSubview(previews[2])
+        
+        rightPreviewsContainerView.spacing = 0
+        rightPreviewsContainerView.axis = .vertical
+        rightPreviewsContainerView.distribution = .equal
+        rightPreviewsContainerView.alignment = .fill
+        previewsContainerView.addArrangedSubview(rightPreviewsContainerView)
+        
+        rightPreviewsContainerView.addArrangedSubview(previews[1])
+        rightPreviewsContainerView.addArrangedSubview(previews[3])
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery.swift:70]: nit: some fancy ASCII art explaining what are the indices of the views within the grid would be cool

File: Sources/StreamChatUI/ChatMessageList/Attachments/GalleryAttachmentViewInjector.swift [hunk 5]
```diff
@@ -36,14 +25,17 @@ public class _GalleryAttachmentViewInjector<ExtraData: ExtraDataTypes>: _Attachm
...
+        galleryView.content = imageAttachments
+        galleryView.didTapOnAttachment = { [weak self] attachment in
+            self?.handleTapOnAttachment(attachment)
+        }
     }
 
-    @objc open func handleTapOnImage() {
-        guard let attachment = imageAttachments.first,
-              let indexPath = contentView.indexPath?()
+    open func handleTapOnAttachment(_ attachment: _ChatMessageAttachment<AttachmentImagePayload>) {
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/Attachments/GalleryAttachmentViewInjector.swift:36]: `_ChatMessageAttachment<AttachmentImagePayload>` -> `ChatMessageImageAttachment`

**Review by VojtaStavik — APPROVED** [review]
LGTM 👍

File: Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery+ImagePreview.swift (hunk 3 above)
Comment (fortmarek) [reply Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery+ImagePreview.swift:74]: the `imageView` does not have appropriate size but calling `imageView.layoutIfNeeded` fixes that

**Pull request merged**